examples/manifold/plot_compare_methods.py

```diff
@@ -48,72 +49,33 @@
 ax.scatter(X[:, 0], X[:, 1], X[:, 2], c=color, cmap=plt.cm.Spectral)
 ax.view_init(4, -72)
 
-methods = ['standard', 'ltsa', 'hessian', 'modified']
-labels = ['LLE', 'LTSA', 'Hessian LLE', 'Modified LLE']
-
-for i, method in enumerate(methods):
+# Set-up manifold methods
+LLE = partial(manifold.LocallyLinearEmbedding,
+              n_neighbors, n_components, eigen_solver='auto')
+
+methods = dict()
+methods['LLE'] = LLE(method='standard')
+methods['LTSA'] = LLE(method='ltsa')
+methods['Hessian LLE'] = LLE(method='hessian')
+methods['Modified LLE'] = LLE(method='modified')
+methods['Isomap'] = manifold.Isomap(n_neighbors, n_components)
+methods['MDS'] = manifold.MDS(n_components, max_iter=100, n_init=1)
+methods['SE'] = manifold.SpectralEmbedding(n_components=n_components,
+                                           n_neighbors=n_neighbors)
+methods['t-SNE'] = manifold.TSNE(n_components=n_components, init='pca',
+                                 random_state=0)
+
+# Plot results
+for i, (label, method) in enumerate(methods.items()):
     t0 = time()
-    Y = manifold.LocallyLinearEmbedding(n_neighbors, n_components,
-                                        eigen_solver='auto',
-                                        method=method).fit_transform(X)
+    Y = method.fit_transform(X)
     t1 = time()
-    print("%s: %.2g sec" % (methods[i], t1 - t0))
-
-    ax = fig.add_subplot(252 + i)
+    print("%s: %.2g sec" % (label, t1 - t0))
+    ax = fig.add_subplot(2, 5, 2 + i + (i > 3))
```

Comment:
I think we should move to the more modern plt.subplots and iterate over the array of axes while we are cleaning up.

Reply:
Yes, I agree and adjusted the axes accordingly. Since the figure includes one 3d axis I had to [remove and re-add that axis](https://stackoverflow.com/questions/35209489/can-i-turn-an-existing-ax-object-into-a-3d-projection) or is there another solution?

Reply:
Forgot about this. Don't know, sorry

Reply:
Okay, no problem, then let's keep the former axes definitions.